bots/psuteam7bot/prophet.js

```diff
@@ -37,183 +55,171 @@ prophet.doAction = (self) => {
             self.log("Base defenders = " + JSON.stringify(nearbyDefenders.length) + ", Assigned as an attacker");
+            self.squadSize = 4;
             self.role = "ATTACKER";
         }
     }
 
-    //Main behavior of Prophet units
     if(self.role === "DEFENDER")
+        return prophet.takeDefenderAction(self);
+
+    if(self.role === "ATTACKER")
+        return prophet.takeAttackerAction(self);
+
+    //Should not fall through unless still UNASSIGNED or something horrible happened
+    self.log('prophet ' + self.role + ' ' + self.me.id + ' still UNASSIGNED!!!')
+    return;
+}
+
+//DEFENDER Behavior
+prophet.takeDefenderAction = (self) =>  {
+    self.log("DEFENDER prophet " + self.id + " taking turn");
+    
+    //Limited movement towards enemy castle
+    if(self.me.turn < 5)
     {
-        self.log("DEFENDER prophet " + self.id + " taking turn");
-        if(self.target === null)
+        if(self.path.length === 0)
         {
-            self.log("prophet finding local defenders")
-            //TODO Change with addition of communication maybe have base record the number of defenders it built and have the prophet receive message from castle
-            //Naive method of just filtering nearby prophets from base location, if passing by ATTACKER strays inside, messes with it
-            const nearbyDefenders = self.getVisibleRobots().filter((robotElement) => {
-                if(robotElement.team === self.me.team && robotElement.unit === self.me.unit && self.me.id != robotElement.id)
-                {
-                    const distance = movement.getDistance(self.base, robotElement);
-                    return distance <= 16;
-                }
-            });
-
-            //general North/East/South/West guard post direction for checking for horizontal/ vertical movement
-            //N = 0
-            //E = 1
-            //S = 2
-            //W = 3
-            const compass = nearbyDefenders.length % 4;
-
-            //Get the direction value from directions array
-            let direction = movement.directions[compass*2];
-
-            let x = 1;  //movement amount for North East
-            let y = 3;
-
-            //If first four rotate left, otherwise rotate right
-            if(nearbyDefenders < 4)
-                direction = movement.rotateDirection(direction, -1);
-            else
-                direction = movement.rotateDirection(direction, 1);
-
-            x *= direction.x;
-            y *= direction.y;
-            self.log('x: ' + x + ' y: ' + y)
-
-            //Case for vertical movement
-            if(compass === 0 || compass === 2)
-                self.target = {x: self.base.x + x, y: self.base.y + y};
-            else
-                self.target = {x: self.base.x + y, y: self.base.y + x};
-
-            self.log("Assigned " + JSON.stringify(self.target) + " as guard post");
+            if(movement.aStarPathfinding(self, self.me, self.potentialEnemyCastleLocation[0], false)) {
```

Comment:
Movement is good, but should add functionality to attack if an enemy is in range first. I.e. if your base is under attack during early turns, defend first and then consider moving to guard post.